sonar-groovy-plugin/src/main/java/org/sonar/plugins/groovy/surefire/GroovySurefireParser.java

```diff
@@ -20,41 +20,44 @@
 package org.sonar.plugins.groovy.surefire;
 
 import com.google.common.collect.Lists;
+
 import org.apache.commons.lang.StringUtils;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-import org.sonar.api.BatchExtension;
-import org.sonar.api.batch.SensorContext;
+import org.sonar.api.batch.BatchSide;
 import org.sonar.api.batch.fs.FilePredicate;
 import org.sonar.api.batch.fs.FilePredicates;
 import org.sonar.api.batch.fs.FileSystem;
 import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.batch.sensor.SensorContext;
 import org.sonar.api.component.ResourcePerspectives;
 import org.sonar.api.measures.CoreMetrics;
 import org.sonar.api.measures.Metric;
 import org.sonar.api.test.MutableTestPlan;
 import org.sonar.api.test.TestCase;
 import org.sonar.api.utils.ParsingUtils;
-import org.sonar.api.utils.SonarException;
-import org.sonar.api.utils.StaxParser;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
 import org.sonar.plugins.groovy.foundation.Groovy;
 import org.sonar.plugins.groovy.surefire.data.SurefireStaxHandler;
 import org.sonar.plugins.groovy.surefire.data.UnitTestClassReport;
 import org.sonar.plugins.groovy.surefire.data.UnitTestIndex;
 import org.sonar.plugins.groovy.surefire.data.UnitTestResult;
+import org.sonar.plugins.groovy.utils.StaxParser;
+import org.sonar.squidbridge.api.AnalysisException;
 
 import javax.xml.stream.XMLStreamException;
 
 import java.io.File;
 import java.io.FilenameFilter;
+import java.io.Serializable;
 import java.util.List;
 import java.util.Map;
 
-public class GroovySurefireParser implements BatchExtension {
+@BatchSide
+public class GroovySurefireParser {
 
-  private static final Logger LOGGER = LoggerFactory.getLogger(GroovySurefireParser.class);
-  private final Groovy groovy;
+  private static final Logger LOGGER = Loggers.get(GroovySurefireParser.class);
   private final ResourcePerspectives perspectives;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this use of "ResourcePerspectives"; it is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)